Src/zipkin4net/Src/Internal/V2SpanConverter.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System;
+using System.Text;
+using zipkin4net.Tracers.Zipkin;
+using zipkin4net.Tracers.Zipkin.Thrift;
+using BinaryAnnotation = zipkin4net.Tracers.Zipkin.BinaryAnnotation;
+using Span = zipkin4net.Tracers.Zipkin.Span;
+
+namespace zipkin4net.Internal
+{
+    internal static class V2SpanConverter
+    {
+        public static Span ToSpan(V2.Span v2Span)
+        {
+            var traceContext = new SpanState(
+                long.Parse(v2Span.TraceId.Substring(0, 8)),
+                long.Parse(v2Span.TraceId.Substring(8, 8)),
+                v2Span.ParentId != null ? long.Parse(v2Span.ParentId) : (long?)null,
+                long.Parse(v2Span.Id),
+                true,
+                v2Span.Debug ?? false);
+
+            var v1Span = new Span(traceContext, v2Span.Timestamp);
+            v1Span.ServiceName = v2Span.LocalServiceName;
+            v1Span.Name = v2Span.Name;
+            v1Span.Endpoint = v2Span.LocalEndpoint.IpEndPoint;
+            v1Span.ServiceName = v2Span.LocalServiceName;
+
+            var startTs = v2Span.Timestamp;
+            var endTs = v2Span.Duration == 0L ? default(DateTime) : v2Span.Timestamp.AddMilliseconds((double) v2Span.Duration / 1000.0);
+            var kind = v2Span.Kind;
+            ZipkinAnnotation
+                cs = null, sr = null, ss = null, cr = null, ms = null, mr = null, ws = null, wr = null;
+            string remoteEndpointType = null;
+            var wroteEndpoint = false;
+
+            foreach (var annotation in v2Span.Annotations)
+            {
+                var annotationValue = annotation.Value;
+                var v1Annotation = new ZipkinAnnotation(annotation.Timestamp, annotationValue);
+                if (annotationValue.Length == 2)
+                {
+                    switch (annotationValue)
+                    {
+                        case zipkinCoreConstants.CLIENT_SEND:
+                            cs = v1Annotation;
+                            remoteEndpointType = zipkinCoreConstants.SERVER_ADDR;
+                            kind = V2.Span.SpanKind.CLIENT;
+                            break;
+                        case zipkinCoreConstants.SERVER_RECV:
+                            sr = v1Annotation;
+                            remoteEndpointType = zipkinCoreConstants.CLIENT_ADDR;
+                            kind = V2.Span.SpanKind.SERVER;
+                            break;
+                        case zipkinCoreConstants.CLIENT_RECV:
+                            kind = V2.Span.SpanKind.CLIENT;
+                            cr = v1Annotation;
+                            break;
+                        case zipkinCoreConstants.MESSAGE_SEND:
+                            kind = V2.Span.SpanKind.PRODUCER;
+                            ms = v1Annotation;
```

Comment:
remoteEndpointType = zipkinCoreConstants.MESSAGE_ADDR;

Reply:
I agree, it should also be corrected here: https://github.com/openzipkin/zipkin/blob/master/zipkin/src/main/java/zipkin/internal/V2SpanConverter.java#L384